lib/storage.js

```diff
@@ -286,6 +286,9 @@ api.insert = async ({edvId, doc}) => {
   if(doc.sequence < 0) {
     throw new TypeError('"doc.sequence" must be a non-negative integer.');
   }
+  if(doc.sequence >= Number.MAX_SAFE_INTEGER) {
```

Comment:
I think you'll want to use `isSafeInteger` for this determiniation.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger

Javascript effectively loses its mind if a value exceed MAX_SAFE_INTEGER so  I don't think it can reliably make comparisons and reason about those values.

Under the covers I suspect this helper is something like this which tests to ensure that the value is LESS THAN or equal to MAX_SAFE_INTEGER

http://speakingjs.com/es5/ch11.html#safe_integers

in addition to all of that, MAX_SAFE_INTEGER itself should be considered a valid value which you will get for free by using `isSafeInteger`.  Even if the existing code functioned reliably it would be rejecting `doc.sequence = Number.MAX_SAFE_INTEGER`.

Reply:
Per our conversation during office hours, I added in the check for both isSafeInteger and less than MAX_SAFE_INTEGER.